### PR TITLE
Add generate-fakes to /tkg/Makefile

### LINKED ISSUE #4500: The /tkg/Makefile does not support `generate-fakes`
**Bug description**

The /tkg/Makefile does not support `generate-fakes` yet this is necessary as the `/tkg/fakes` directory exists and was generated via the [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter) mod used in the root `Makefile`.

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by benjaminapetersen**
### What this PR does / why we need it

The `/tkg` directory contains `/tkg/fakes` which are generated from various interfaces such as `/tkg/client/client.go`, etc.  The `make generate-fakes` command exists in the root `Makefile` but should also exist in the `/tkg/Makefile` to support the generation of fakes.

### Which issue(s) this PR fixes

#4500 

Fixes #

### Describe testing done for PR

Ran `make generate-fakes` in the `/tkg` directory.

### Release note

```release-note

```

### CODE REVIEW

**Review by joshuatcasey — APPROVED** [review]

**Comment by benjaminapetersen** [comment]
@vmware-tanzu/tanzu-framework-reviewers best guess

**Review by prkalle — APPROVED** [review]
LGTM with one minor comment. Thanks!

File: tkg/Makefile [hunk 1]
```diff
@@ -63,3 +63,21 @@ tools: $(TOOLING_BINARIES) ## Build tooling binaries
 $(TOOLING_BINARIES):
 	make -C $(TOOLS_DIR) $(@F)
 
+
+
+## --------------------------------------
+##@ Generate files
+## --------------------------------------
+
+# .PHONY: generate-fakes
```
[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply tkg/Makefile:72]: Makes for some nice headings when `make` is run w/o arguments, but I'm not sure if this `Makefile` will grow a ton over time?

Screen Shot 2023-03-21 at 1 14 48 PM

[the remark this answers is not in the document]
Comment (prkalle) [reply tkg/Makefile:72]: The help comes from the text after the double# in the below line. 
```
generate-fakes: ## Generate fakes for writing unit tests
```

But Line L72-L75 is commented make target which can be removed

**Comment by benjaminapetersen** [comment]
I did not expect `lint` to be flaky.  Good to know...

**Pull request merged**